key/keys.go

```diff
@@ -76,20 +80,28 @@ func (i *Identity) Equal(i2 *Identity) bool {
 }
 
 // SelfSign signs the public key with the key pair
-func (p *Pair) SelfSign() {
+func (p *Pair) SelfSign() error {
 	msg := p.Public.Hash()
-	signature, _ := AuthScheme.Sign(p.Key, msg)
+	signature, err := p.Public.Scheme.AuthScheme.Sign(p.Key, msg)
+	if err != nil {
+		return err
+	}
 	p.Public.Signature = signature
+	return nil
 }
 
 // NewKeyPair returns a freshly created private / public key pair. The group is
 // decided by the group variable by default.
 func NewKeyPair(address string) *Pair {
```

Comment:
Why not passing the scheme in argument ? + change comments which are outdated now (relating to "by default")